### DISCUSSION TITLE
Populate properties using both query strings and route params

### DISCUSSION THREAD

**Discussion by dorinniscu** [livewire/livewire#3335] [7. Bugs (archived)]
I use a dropdown to filter the results by category.

When I use **query params** and reload the page everything works as expected, the category is set correctly. 

**products?category=shirt**

When I use a **route params** and reload the page the category is not applied. (It works when I change the value, it also changes the category in URL) but when I reload the page the value is lost.

**products/shirt**

**Route::get('/products/{category?}', Products::class)->name('products.index');**

```
class Products extends Component
{
    public string $category = '';

    protected $queryString = [
        'category' => ['except' => ''],
    ];
}
```

If I set it inside mount will work fine but I was expecting to bind it directly as query string does.

```
/**
 * Mount.
 *
 * @param  string  $category
 * @return void
 */
public function mount(string $category = ''): void
{
    $this->category= $category;
}
```

**Comment by joshhanley** [comment]
@dorinniscu can you share your component class?

**Reply by dorinniscu** [reply]
@joshhanley Sure. I edited the description with the simplified version related to this behavior.

**Comment by mikethetreeclimber** [comment]
Not sure if this is relevant to your issue but I had a similar issue of values becoming undefined after ajax requests when I was using the mount method and was able to fix the errors by putting the logic from the mount method in the render method.

**Reply by dorinniscu** [reply]
It`s not the case. If I set the value inside mount() works fine but I was expecting to just bind the value (as query string does) without doing it manually.

**Reply by mikethetreeclimber** [reply]
So you are trying to pass the query string to the Route from your components class? 

So this approach wont give a different result?  I am still relatively new to livewire so I am also trying to understand more about it
```
public function render()
{
$queryString = [
        'category' => ['except' => ''],
    ];
return view('some-view', compact($queryString));  
}
```
